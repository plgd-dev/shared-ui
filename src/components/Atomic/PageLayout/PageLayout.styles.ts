import { css } from '@emotion/react'
import { ThemeType, get } from '../_theme'

export const pageLayout = css`
    height: 100%;
    display: flex;
    flex-direction: column;
`

export const top = css`
    padding: 24px 0 0 0;
    height: 100%;
    display: flex;
    flex-direction: column;
    flex: 1 1 auto;
    overflow: hidden;
`

export const header = css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-bottom: 16px;
    min-height: 72px;
`

export const headerBorder = (theme: ThemeType) => css`
    position: relative;

    &:after {
        content: '';
        display: block;
        position: absolute;
        left: 40px;
        right: 40px;
        bottom: 0;
        width: calc(100% - 80px);
        height: 1px;
        background: ${get(theme, 'PageLayout.headline.border.color')};
    }
`

export const padding = css`
    padding-left: 40px;
    padding-right: 40px;
`

export const contentPadding = css`
    height: 100%;
`

export const left = css`
    display: flex;
    align-items: center;
    flex: 1 1 auto;
    position: relative;
    z-index: 1;
`

export const headline = (theme: ThemeType) => css`
    color: ${get(theme, 'PageLayout.headline.color')};
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 400px;
    white-space: nowrap;
`

export const statusTag = css`
    padding-left: 8px;
`

export const rightActions = css``
