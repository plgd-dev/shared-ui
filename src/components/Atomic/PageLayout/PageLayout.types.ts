import { ReactNode } from 'react'

export type Props = {
    breadcrumbs?: any
    children?: any
    dataTestId?: string
    footer?: ReactNode
    header?: any
    headlineStatusTag?: ReactNode
    loading?: any
    title?: string
}
