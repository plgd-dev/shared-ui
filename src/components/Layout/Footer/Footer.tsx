import { FC, memo } from 'react'
import { motion } from 'framer-motion'
import isFunction from 'lodash/isFunction'
import { Props } from './Footer.types'
import * as styles from './Footer.styles'
import { Icon } from '../../Atomic/Icon'

const Footer: FC<Props> = memo((props) => {
    const { paginationComponent, recentTasksPortal, recentTasksPortalTitle, footerExpanded, setFooterExpanded } = props

    const handleToggle = (e: any) => {
        e.preventDefault()
        isFunction(setFooterExpanded) && setFooterExpanded(!footerExpanded)
    }

    const LeftSide = () => {
        if (recentTasksPortalTitle) {
            return (
                <a css={styles.recentTasks} href='packages/shared-ui/src/components/Layout/Footer#' onClick={handleToggle}>
                    {recentTasksPortalTitle} <Icon css={[styles.icon, footerExpanded && styles.panelOpen]} icon='arrow-triangle-full-up' />
                </a>
            )
        } else {
            return <div />
        }
    }

    return (
        <motion.div
            layout
            animate={{
                height: footerExpanded ? 420 : 65,
                flex: `0 0 ${footerExpanded ? 420 : 65}`,
            }}
            css={styles.footer}
            exit={{
                height: 65,
            }}
            initial={false}
            transition={{
                duration: 0.3,
            }}
        >
            <div css={styles.footerInner}>
                <div css={styles.footerMainLine}>
                    <LeftSide />
                    <div>{paginationComponent}</div>
                </div>

                {recentTasksPortal}
            </div>
        </motion.div>
    )
})

Footer.displayName = 'Footer'

export default Footer
