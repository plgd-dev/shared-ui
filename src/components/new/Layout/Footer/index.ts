export { default } from './Footer'
export * from './Footer'

export { default as Pagination } from './Pagination'
export * from './Pagination'
