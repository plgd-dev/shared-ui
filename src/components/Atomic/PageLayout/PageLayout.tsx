import React, { forwardRef } from 'react'

import { Props, defaultProps } from './PageLayout.types'
import * as styles from './PageLayout.styles'
import Headline from '../Headline'
import PageLoader from '../PageLoader'
import ConditionalWrapper from '../ConditionalWrapper'
import { useDocumentTitle } from '../../../common/hooks'

const PageLayout = forwardRef<HTMLDivElement, Props>((props, ref) => {
    const { children, dataTestId, headlineStatusTag, title, header, headerBorder, footer, loading, collapsed, xPadding } = {
        ...defaultProps,
        ...props,
    }

    useDocumentTitle(title || '')

    return (
        <div css={styles.pageLayout} ref={ref}>
            <div css={styles.top}>
                <PageLoader className='auth-loader' collapsed={collapsed} loading={loading} />
                <div css={[styles.header, styles.padding, headerBorder && styles.headerBorder]}>
                    <div css={styles.left}>
                        <Headline css={styles.headline} dataTestId={dataTestId?.concat('-title')} type='h4'>
                            {title}
                        </Headline>
                        {headlineStatusTag && <div css={styles.statusTag}>{headlineStatusTag}</div>}
                    </div>
                    <div css={styles.rightActions}>{header}</div>
                </div>
                <ConditionalWrapper condition={xPadding === true} wrapper={(c) => <div css={[styles.padding, styles.contentPadding]}>{c}</div>}>
                    {children}
                </ConditionalWrapper>
            </div>
            {footer}
        </div>
    )
})

PageLayout.displayName = 'PageLayout'

export default PageLayout
