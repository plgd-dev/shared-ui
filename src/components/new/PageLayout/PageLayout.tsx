import React, { FC } from 'react'
import { Props } from './PageLayout.types'
import * as styles from './PageLayout.tsx.styles'
import Headline from '../Headline'
import { Helmet } from 'react-helmet'

const PageLayout: FC<Props> = (props) => {
    const { children, headlineStatusTag, title, header } = props
    return (
        <div css={styles.pageLayout}>
            <Helmet>
                <title>{title}</title>
            </Helmet>
            <div css={styles.header}>
                <div css={styles.left}>
                    <Headline css={styles.headline} type='h4'>
                        {title}
                    </Headline>
                    {headlineStatusTag && <div css={styles.statusTag}>{headlineStatusTag}</div>}
                </div>
                <div css={styles.rightActions}>{header}</div>
            </div>
            {children}
        </div>
    )
}

PageLayout.displayName = 'PageLayout'

export default PageLayout
