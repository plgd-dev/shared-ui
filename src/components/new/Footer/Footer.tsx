import { FC, memo, useState } from 'react'
import { useIntl } from 'react-intl'
import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
import Tooltip from 'react-bootstrap/Tooltip'
import Popover from 'react-bootstrap/Popover'
import { format } from 'date-fns'
import { messages as t } from './Footer.i18n'
import './Footer.scss'
import { useAppConfig } from '@/containers/App'
import { copyToClipboard } from '../../../common/utils'
import { showSuccessToast } from '../../old/toast'
import Display from '../Display'
import { FooterLinkType, Props } from './Footer.types'

const Footer: FC<Props> = memo(({ links }) => {
    const { formatMessage: _ } = useIntl()
    const { buildInformation } = useAppConfig()
    const [show, setShow] = useState(false)

    const handleOnMouseEnter = () => {
        setShow(true)
    }
    const handleOnMouseLeave = () => {
        setShow(false)
    }

    const copyVersion = () => {
        if (buildInformation?.version) {
            copyToClipboard(buildInformation.version)
            showSuccessToast({
                title: _(t.done),
                message: _(t.copied),
            })
        }
    }

    const BuildInformation = () => {
        if (!buildInformation?.buildDate) {
            return null
        }
        return (
            <Display when={!!buildInformation}>
                <OverlayTrigger
                    placement='top'
                    overlay={
                        <Tooltip className='plgd-tooltip' id='footer-version'>
                            {_(t.clickToCopy)}
                        </Tooltip>
                    }
                >
                    <span className='copy' onClick={copyVersion}>
                        <i className={`fas fa-copy m-r-10`} />
                    </span>
                </OverlayTrigger>

                <OverlayTrigger
                    placement='top-start'
                    show={show}
                    delay={{ show: 0, hide: 1000 }}
                    overlay={
                        <Popover
                            id='popover-positioned-top'
                            className='plgd-popover interactive'
                            onMouseEnter={handleOnMouseEnter}
                            onMouseLeave={handleOnMouseLeave}
                        >
                            <Popover.Body>
                                <div className='footerBuildInfo'>
                                    <div className='line'>
                                        <div className='attr'>Build time:</div>
                                        <div className='val'>{format(new Date(buildInformation.buildDate), 'Pp')}</div>
                                    </div>
                                    <div className='line'>
                                        <div className='attr'>Commit hash:</div>
                                        <a
                                            className='val'
                                            href={`https://github.com/plgd-dev/client-application/commit/${buildInformation.commitHash}`}
                                            target='_blank'
                                        >
                                            {buildInformation.commitHash}
                                        </a>
                                    </div>
                                </div>
                            </Popover.Body>
                        </Popover>
                    }
                >
                    <span onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave}>
                        {buildInformation.version}
                    </span>
                </OverlayTrigger>
            </Display>
        )
    }

    return (
        <footer id='footer'>
            <div id='pending-commands-wrapper'></div>
            <div className='left'>
                <BuildInformation />
            </div>
            <div className='right'>
                {links.map((item: FooterLinkType, index) => {
                    // @ts-ignore
                    const tran = _(t[item.i18key])
                    return (
                        <a href={item.to} target='_blank' rel='noopener' key={index}>
                            {tran}
                        </a>
                    )
                })}
            </div>
        </footer>
    )
})

Footer.displayName = 'Footer'

export default Footer
