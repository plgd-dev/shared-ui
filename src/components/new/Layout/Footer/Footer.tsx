import { FC, memo } from 'react'
import { Props } from './Footer.types'
import * as styles from './Footer.styles'
import { Icon } from '../../Icon'
import { motion } from 'framer-motion'

const Footer: FC<Props> = memo((props) => {
    const { paginationComponent, recentTasksPortal, recentTasksPortalTitle, footerExpanded, setFooterExpanded } = props

    const handleToggle = (e: any) => {
        e.preventDefault()
        setFooterExpanded(!footerExpanded)
    }

    const LeftSide = () => {
        if (recentTasksPortalTitle) {
            return (
                <a css={styles.recentTasks} href='#' onClick={handleToggle}>
                    {recentTasksPortalTitle} <Icon css={[styles.icon, footerExpanded && styles.panelOpen]} icon='arrow-triangle-full-up' />
                </a>
            )
        } else {
            return <div />
        }
    }

    return (
        <motion.div
            layout
            animate={{
                height: footerExpanded ? 420 : 65,
                flex: `0 0 ${footerExpanded ? 420 : 65}`,
            }}
            css={styles.footer}
            exit={{
                height: 65,
            }}
            initial={false}
            transition={{
                duration: 0.3,
            }}
        >
            <div css={styles.footerMainLine}>
                <LeftSide />
                <div>{paginationComponent}</div>
            </div>

            {recentTasksPortal}
        </motion.div>
    )
})

Footer.displayName = 'Footer'

export default Footer
