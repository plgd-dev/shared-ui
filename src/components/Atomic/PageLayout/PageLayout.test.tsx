import React from 'react'
import { render } from '@testing-library/react'
import PageLayout from './PageLayout'
import Footer from '../../Layout/Footer'

describe('<PageLayout>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <PageLayout
                footer={
                    <Footer
                        footerExpanded={false}
                        paginationComponent={<div id='paginationPortalTarget'></div>}
                        recentTasksPortal={<div id='recentTasksPortalTarget'></div>}
                        recentTasksPortalTitle={<span>recentTasksPortalTitle</span>}
                    />
                }
                header={<div>Custom Header</div>}
                loading={false}
                title='PageLayoutTitle'
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
