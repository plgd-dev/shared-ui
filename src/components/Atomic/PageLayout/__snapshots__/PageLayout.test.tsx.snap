// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  padding: 24px 0 0 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 16px;
  min-height: 72px;
  padding-left: 40px;
  padding-right: 40px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
  z-index: 1;
}

.emotion-4 {
  font-weight: 700;
  font-style: normal;
  font-size: 24px;
  line-height: 30px;
  letter-spacing: -0.5px;
  margin: 0;
}

@media (max-width: 767px) {
  .emotion-4 {
    font-size: 20px;
    line-height: 24px;
  }
}

.emotion-6 {
  padding-left: 40px;
  padding-right: 40px;
  height: 100%;
}

.emotion-7 {
  border-top: 1px solid;
  padding: 0 40px;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
}

.emotion-8 {
  -webkit-flex: 0 0 420px;
  -ms-flex: 0 0 420px;
  flex: 0 0 420px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 16px 40px;
  min-height: 64px;
  margin: 0 -40px;
}

.emotion-10 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 18px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 22px;
  position: relative;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.emotion-11 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 0;
  color: #81868C;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <h4
            class="h4 emotion-4"
          >
            PageLayoutTitle
          </h4>
        </div>
        <div
          class="emotion-5"
        >
          <div>
            Custom Header
          </div>
        </div>
      </div>
      <div
        class="emotion-6"
      />
    </div>
    <div
      class="emotion-7"
      style="height: 65px;"
    >
      <div
        class="emotion-8"
      >
        <div
          class="emotion-9"
        >
          <a
            class="emotion-10"
            href="#"
          >
            <span>
              recentTasksPortalTitle
            </span>
             
            <svg
              class="emotion-11"
              fill="none"
              height="16"
              viewBox="0 0 6 11"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M.126 6.854a.387.387 0 0 0 .606 0L3 4.207l2.268 2.647a.387.387 0 0 0 .606 0 .56.56 0 0 0 0-.708l-2.571-3a.387.387 0 0 0-.606 0l-2.571 3a.56.56 0 0 0 0 .708"
                fill="currentcolor"
                fill-rule="evenodd"
              />
              <g
                clip-path="url(#icon-arrow-triangle-full-up_svg__a)"
              >
                <path
                  d="M2.576 3.196a.56.56 0 0 1 .85 0l2.4 2.666a.72.72 0 0 1 .129.727c-.094.25-.311.413-.555.413L.6 7c-.242 0-.461-.163-.555-.413a.72.72 0 0 1 .13-.727l2.4-2.666z"
                  fill="currentcolor"
                />
              </g>
              <defs>
                <clippath
                  id="icon-arrow-triangle-full-up_svg__a"
                >
                  <path
                    d="M0 11h6V.333H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </a>
          <div>
            <div
              id="paginationPortalTarget"
            />
          </div>
        </div>
        <div
          id="recentTasksPortalTarget"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
